codemp/game/g_public.h

```diff
@@ -254,6 +254,86 @@ typedef struct sharedEntity_s {
 	int				next_roff_time; //rww - npc's need to know when they're getting roff'd
 } sharedEntity_t;
 
+typedef struct sharedEntity_qvm_s {
+	entityState_t	s;				// communicated by server to clients
+	uint32_t		playerState;	//needs to be in the gentity for bg entity access
+									//if you want to actually see the contents I guess
+									//you will have to be sure to VMA it first.
+	uint32_t		m_pVehicle; //vehicle data
+	uint32_t		ghoul2; //g2 instance
+	int				localAnimIndex; //index locally (game/cgame) to anim data for this skel
+	vec3_t			modelScale; //needed for g2 collision
+
+	//from here up must also be unified with bgEntity/centity
+
+	entityShared_t	r;				// shared by both the server system and game
+
+	//Script/ICARUS-related fields
+	int				taskID[NUM_TIDS];
+	uint32_t		parms;
+	uint32_t		behaviorSet[NUM_BSETS];
+	uint32_t		script_targetname;
+	int				delayScriptTime;
+	uint32_t		fullName;
+
+	//rww - targetname and classname are now shared as well. ICARUS needs access to them.
+	uint32_t		targetname;
+	uint32_t		classname;			// set in QuakeEd
+
+	//rww - and yet more things to share. This is because the nav code is in the exe because it's all C++.
+	int				waypoint;			//Set once per frame, if you've moved, and if someone asks
+	int				lastWaypoint;		//To make sure you don't double-back
+	int				lastValidWaypoint;	//ALWAYS valid -used for tracking someone you lost
+	int				noWaypointTime;		//Debouncer - so don't keep checking every waypoint in existance every frame that you can't find one
+	int				combatPoint;
+	int				failedWaypoints[MAX_FAILED_NODES];
+	int				failedWaypointCheckTime;
+
+	int				next_roff_time; //rww - npc's need to know when they're getting roff'd
+} sharedEntity_qvm_t;
+
+typedef struct sharedEntityMapper_s {
+	entityState_t	*s;				// communicated by server to clients
```

Comment:
if this used references (or some kind of dumb smart pointer?) instead of raw pointers, we could avoid having to turn all the `.` into `->`, which would make this more appealing to merge downstream. But I've not looked at enough of the context to know if that's possible; particularly whether references' initialize-once-and-never-reassign semantics work here.

Reply:
Actually, I don't think you can overload `operator.`, so references are probably the only way to avoid adding another level of pointer indirection.